apps/authentication-service/src/shared/tokenUtils.test.ts

```diff
@@ -0,0 +1,144 @@
+import { getTestToken } from "@/test/testData/users";
+import fs from "fs";
+
+import {
+  getCachedToken,
+  getIsTokenExpired,
+  getLocalToken,
+  logToken,
+  setCachedToken,
+  setLocalToken,
+} from "@/shared/tokenUtils";
+
+let tokenBackup: string;
+const validTestToken: string = getTestToken();
+const expiredTestToken: string = getTestToken(true);
+
+describe("tokenUtils", () => {
+  beforeAll(() => {
+    tokenBackup = getCachedToken() || "";
+  });
+
+  afterAll(() => {
+    if (tokenBackup) {
+      setCachedToken(tokenBackup);
+    }
+    jest.clearAllMocks();
+  });
+
+  describe("setCachedToken", () => {
+    it("sets token", () => {
+      jest.spyOn(fs, "existsSync").mockReturnValue(true);
+      jest.spyOn(fs, "writeFileSync").mockImplementation(() => {});
+      jest.spyOn(fs, "readFileSync").mockReturnValue(validTestToken);
+      setCachedToken(validTestToken);
+      expect(getCachedToken()).toBe(validTestToken);
+    });
+
+    it("throws an error when unable to cache token", () => {
```

Comment:
This it statement is what we would actually want, but because the function swallows errors it just tests that it returns false